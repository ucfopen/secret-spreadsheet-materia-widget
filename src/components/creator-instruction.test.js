import React from 'react'
import Instruction from './creator-instruction'
import renderer from 'react-test-renderer'
import { shallow } from '../enzyme'

const makeProps = () => {
  return({
    toggleInstruction: jest.fn(),
    toggleKeyboardInst: jest.fn(),
  })
}

describe('CreatorInstruction component', function() {

	beforeEach(() => {
		jest.resetModules()
  })

  test('CreatorInstruction toggles keyboard control with onClick', () => {
		const props = makeProps()

    const component = renderer.create(<Instruction {... props}/>)
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot()
	})

	test('CreatorInstruction toggles keyboard control with onClick', () => {
		const props = makeProps()

		const component = shallow(<Instruction {... props}/>)

		component.find('.keyboard-controls-div').simulate('Click')
		expect(props.toggleKeyboardInst).toBeCalled()
	})

	test('CreatorInstruction toggles keyboard control with Enter keyPress', () => {
		const props = makeProps()

		const component = shallow(<Instruction {... props}/>)

		component.find('.keyboard-controls-spam').simulate('keypress', {key: 'Enter'})
		expect(props.toggleKeyboardInst).toBeCalled()
	})

	test('CreatorInstruction toggles keyboard control with non-Enter keyPress', () => {
		const props = makeProps()

		const component = shallow(<Instruction {... props}/>)

		component.find('.keyboard-controls-spam').simulate('keypress', {key: 'a'})
		expect(props.toggleKeyboardInst).not.toBeCalled()
	})


	test('CreatorInstruction calls toggleInstruction with Enter key', () => {
		const props = makeProps()

		const component = shallow(<Instruction {... props}/>)
		component.find('.close').simulate('keypress', {key: 'Enter'})
		expect(props.toggleInstruction).toBeCalled()
	})

  test('CreatorInstruction fails to call toggleInstruction with non-Enter keyPress', () => {
		const props = makeProps()

		const component = shallow(<Instruction {... props}/>)
		component.find('.close').simulate('keypress', {key: 'a'})
		expect(props.toggleInstruction).not.toBeCalled()
	})

})
