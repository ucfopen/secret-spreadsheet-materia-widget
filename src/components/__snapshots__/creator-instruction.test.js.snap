// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatorInstruction component CreatorInstruction toggles keyboard control with onClick 1`] = `
<div
  className="table-text instruction-hidden"
>
  <span
    className="close"
    onClick={[MockFunction]}
    onKeyPress={[Function]}
    tabIndex={0}
  >
    ×
  </span>
  <h2>
    WHAT TO DO
  </h2>
  <ul
    className="what-to-do"
  >
    <li>
      Add rows and columns, then input data in the cells below.
    </li>
    <li
      className="list-item-hidden"
    >
      Check cells to turn them 
      <span
        className="blue-text"
      >
        blue
      </span>
       - these will be left blank for students to fill out.
    </li>
    <li
      className=""
    >
      The widget will automatically hide the given number of cells
    </li>
    <li
      className="keyboard-controls-div"
      onClick={[MockFunction]}
    >
      <span
        className="keyboard-controls-spam"
        onKeyPress={[Function]}
        tabIndex={0}
      >
        Keyboard controls
      </span>
      
    </li>
  </ul>
</div>
`;
